src/Symfony/Component/Config/Loader/FileLoader.php

```diff
@@ -86,7 +86,12 @@ public function import($resource, $type = null, $ignoreErrors = false, $sourceRe
                 // we fallback to the current locator to keep BC
                 // as some some loaders do not call the parent __construct()
                 // @deprecated should be removed in 3.0
-                $locator = $loader->getLocator() ?: $this->locator;
+                $locator = $loader->getLocator();
+                if (null === $locator) {
+                    @trigger_error('Not calling the parent constructor to initialize the locator in classes extending FileLoader is deprecated since version 2.7 and will be removed in 3.0.', E_USER_DEPRECATED);
```

Comment:
Here an example: `Not calling the parent constructor in Symfony\Component\Config\Tests\Loader\TestFileLoader which extends Symfony\Component\Config\Loader\FileLoader is deprecated since version 2.7 and will not be supported anymore in 3.0.`